packages/core/src/components/editable-text/editableText.tsx

```diff
@@ -228,11 +237,27 @@ export class EditableText extends React.Component<IEditableTextProps, IEditableT
         safeInvoke(this.props.onChange, value);
     };
 
-    private handleKeyEvent = ({ ctrlKey, metaKey, which }: React.KeyboardEvent<HTMLElement>) => {
-        if (which === Keys.ENTER && (!this.props.multiline || ctrlKey || metaKey)) {
-            this.toggleEditing();
-        } else if (which === Keys.ESCAPE) {
+    private handleKeyEvent = (event: React.KeyboardEvent<HTMLElement>) => {
+        const { altKey, ctrlKey, metaKey, shiftKey, which } = event;
+        if (which === Keys.ESCAPE) {
             this.cancelEditing();
+            return;
+        }
+
+        const groupKeys = altKey || ctrlKey || metaKey || shiftKey;
+        if (which === Keys.ENTER) {
+            // prevent IE11 from full screening with alt + enter
+            // shift + enter adds a newline by default
+            if (altKey || shiftKey) { event.preventDefault(); }
+
+            if (this.props.shouldConfirmOnEnter && this.props.multiline) {
+                if (event.target != null && groupKeys) {
+                    insertAtCaret(event.target as HTMLTextAreaElement, "\n");
+                    this.handleTextChange(event);
+                } else { this.toggleEditing(); }
```

Comment:
nit: I prefer to kick one-line if/else blocks to a separate line to reduce line density and to promote readability by maintaining a consistent visual pattern, so this would be:

```
// ...
} else {
  this.toggleEditing();
}
```